atom/browser/api/lib/web-contents.coffee

```diff
@@ -79,6 +93,39 @@ wrapWebContents = (webContents) ->
 
     @_printToPDF printingSetting, callback
 
+  webContents.session.clearStorageData = (options, callback) ->
+    if not callback? and typeof options is 'function'
+      callback = options
+      options = null
+    options ?= origin: ''
+    options.origin ?= ''
+    options.begin ?= new Date(0)
+    options.end ?= new Date()
+    options.storageType ?= []
+    options.quotaType ?= ['persistent']
+    throw new TypeError('StorageType need to be array') unless Array.isArray options.storageType
+    throw new TypeError('QuotaType need to be array') unless Array.isArray options.quotaType
+
+    storageType = 0
+    for prop, value of storageMask
+      storageType |= value if prop in options.storageType
+
+    quotaType = 0
+    for prop, value of quotaMask
+      quotaType |= value if prop in options.quotaType
+
+    webContents.session._clearStorageData String(options.origin), storageType, quotaType,
+        options.begin.getTime(), options.end.getTime(), callback
```

Comment:
No, if you really want to add JS wrapper for Session you should follow how WebContents has done it: by saving the wrap function into C++ and call it when creating a Session object in C++. You can see the implementation of `_setWrapWebContents` to have an idea.

And in this case I think we should just do parameter parsing in C++.